### DISCUSSION TITLE
Must have features!

### DISCUSSION THREAD

**Discussion by Ali-Ferwez** [sumatrapdfreader/sumatrapdf#2761] [Ideas]
Read aloud support, dictionary and web search support is required to make this a good option for viewing docs. Right now this is only good for viewing comics and similar.  To reading pdfs Edge browser, Acrobat, Foxit all do better job than Sumatra reader. So developers could please you consider my suggestions? Anyway thanks for good work making this program free and open-source.

**Comment by tactilis** [comment]
> Right now this is only good for viewing comics and similar.

Nonsense. I use SumatraPDF exclusively for reading PDFs.

Search support:
 
![image](https://user-images.githubusercontent.com/4060092/176127127-6454ce57-856b-4d1f-a743-d02c0abf24d9.png)
 
--- 
You could help yourself by reading the excellent Manual (accessible from the Help menu) before making incorrect assertions:
 
![image](https://user-images.githubusercontent.com/4060092/176128098-cdf6e9b0-b941-4d10-b3d0-690b5e926063.png)

**Comment by Ali-Ferwez** [comment]
Yeah, I missed that one. But still two clicks away. Cumbersome for me who do lot of lookups now and then. And still it is one off three features I suggested. If you have used Microsoft Edge inbuilt pdf viewing, You would have noticed that web search is a single click away, also search with bing offers sidebar quick web view in the same tab page. Also nice little crafty feature define offers quick overlay definition still a one click away. But all of which that concerns me about Sumatra Reader most is lack of support for narrator.  It doesn't posses built in reader(TTS) either.

**Reply by GitHubRulesOK** [reply]
I can send current PDF page text to windows SAPI.cmd file to read the page in a couple of clicks, but have never bothered (I did write AHK script to read page selections), since I can tell EDGE to open that page and do a very reasonable job. So why bust a gut with such a wildly variable input? Try read aloud on many pdfs is good for a laugh, and switch off and on and off when the page has garbage text as plain text stored or ocr has got totally the wrong characters.

**Reply by Ali-Ferwez** [reply]
TTS is miserable in OCRed docs, but not for standardized e-publications that meant to work with assistive technologies. If a document adheres to pdf standards then TTS does its charm well. And yes, Edge does a marvelous job reading pdfs but I  would rather not use a web browser as my primary doc reader 'cause obvious privacy concerns. Now I am not complaining but I would prefer if Sumatra reader becomes an all-rounder for reading like 7-Zip for archives. It would be nice to have a single fast concise program to all your e-document reading than several app installs for each format and features.
Adobe Acrobat: good compatibility, online syncing, collaboration, but slow and cumbersome.
Foxit: Fast, adaptive, compatible but with strong Chinese roots, significant portion of software is developed in China though company itself has significant US / international presence.
Browser: Edge, Chrome, Firefox. yeah they are good and convenient but not always. Privacy and ease of dedicated reader is preferred in many cases though it may ultimately come down to personal preferences.

**Reply by GitHubRulesOK** [reply]
Edge privacy is no different to Windows that you lease from a Microsoft Domain controller so yes offline PDF via Foxit/Skia embedded viewer is a potential risk due to JavaScript enabled.

And yes I still trust Sumatra online browsing in PaleMoon (or WaterFox) for exactly the reason it does not allow phone home actions (e.g. no scripted forms, no run this and that) but there are very few good PDF>TTS engines.

 I have found a few generic text/ocr on windows are better than many so previously considered Russian Qtranslate (with Open.OCR) has previously been good but current portable is not downloading so I would need to search around for similar. 
[Dspeech](http://dimio.altervista.org/eng/#DSpeech) is useful lightweight uses PDFtotext but AFAIK not OCR (It's possible to specify a file name to open and automatically reproduce.) and not working in my Win11sandbox

**Reply by Ali-Ferwez** [reply]
If you are interested there is a program called [Balabolka.](http://www.cross-plus-a.com/balabolka.htm) All computer voices installed on your system are available to Balabolka. (SAPI). The on-screen text can be saved as an audio file. The program can read the clipboard content, extract text from documents, customize font and background colour, control reading from the system tray or by the global hotkeys. Supported text file formats: AZW, AZW3, CHM, DjVu, DOC, DOCX, EML, EPUB, FB2, FB3, HTML, LIT, MD, MOBI, ODP, ODS, ODT, PDB, PRC, PDF, PPT, PPTX, RTF, TCR, WPD, XLS, XLSX. Allows to alter a voice's parameters, including rate and pitch. the portable version available [here.](https://portableapps.com/apps/accessibility/balabolka-portable)
Optional secure download of installer at [softonic](https://balabolka.en.softonic.com/)

**Reply by GitHubRulesOK** [reply]
I am still looking for good image reader, this is close but spot the differences (not an audio problem?)
![image](https://user-images.githubusercontent.com/12754913/176280182-220f9765-fcf2-430d-a712-e250f8f9e0e8.png)
result
```
If you are interested there is a program called Balabolka. Al computer voices installed on your system are available to Balabolka.
(SAP). The on-screen text can be saved as an audio file. The program can read the clipboard content, extract text from documents,
customize font and background colour, control reading from the system tray or by the global hotkeys. Supported text file formats:
AZW, AZW3, CHM, DjVu, DOC, DOCX, EML, EPUB, FB2, FB3, HTML, LIT, MD, MOBI, ODP, ODS, ODT, PDB, PRC, PDF, PPT, PPTX, RTE, TCR
WPD, XLS, XLSX. Allows to alter a voice's parameters, including rate and pitch. the portable version available here.
Optional secure download of installer at softonic
```

**Reply by Ali-Ferwez** [reply]
Best OCR I have come across is Acrobat pro, Pretty decent most of the time. There is a app called ["Rossum"](https://rossum.ai/) which is touted for its high accuracy in corporate circles though I have never used it. Those are paid apps, as free apps for this category especially good ones are hard to come by because processing images involves artificial neural networks and deep machine learning at the very high end and involves lot of capital cost for research.
There is one app I have very recently come across which had the backing of Google until recently but Open Source called [Tessaract](https://github.com/tesseract-ocr/tesseract), essentially a CLI, though 3rd party GUI frontends can also be found in [here](https://tesseract-ocr.github.io/tessdoc/User-Projects-%E2%80%93-3rdParty.html) , though again I haven't had the chance to use it, but assumed as very good, since it had corporate backing previously.

**Comment by jakima** [comment]
@GitHubRulesOK I'd suggest trying [dpscreenocr](https://github.com/danpla/dpscreenocr).

**Comment by GitHubRulesOK** [comment]
@jakima 
I dropped CopyFish some time ago (unsure why as its good for some cases) and use NormCap 
![image](https://user-images.githubusercontent.com/12754913/233662117-b21b51d4-f3e9-476b-b2d6-0d495c984e45.png) as very easy to use, especially as I use dropdown launcher, and also use BalCon (Balabolka  for Console) command lines,

For test comment above the result is as perfect as can be expected. **i.e. better than CopyFish**, but may wrap lines differently.

>If you are interested there is a program called Balabolka. All computer voices installed on your system are available to Balabolka. (SAPI). The on-screen text can be saved as an audio file. The program can read the clipboard content, extract text from documents, customize font and background colour, control reading from the system tray or by the global hotkeys. Supported text file formats: AZW, AZW3, CHM, DjVu, DOC, DOCX, EML, EPUB, FB2, FB3, HTML, LIT, MD, MOBI, ODP, ODS, ODT, PDB, PRC, PDF, PPT, PPTX, RTF, TCR, WPD, XLS, XLSX. Allows to alter a voice's parameters, including rate and pitch. the portable version available here.
Optional secure download of installer at softonic

dpscreenOCR, oddly adds a blank line in some cases, but layout is closer to "as seen"

>If you are interested there is a program called Balabolka. All computer voices installed on your system are available to Balabolka.
(SAPI). The on-screen text can be saved as an audio file. The program can read the clipboard content, extract text from documents,
customize font and background colour, control reading from the system tray or by the global hotkeys. Supported text file formats:
AZW, AZW3, CHM, DjVu, DOC, DOCX, EML, EPUB, FB2, FB3, HTML, LIT, MD, MOBI, ODP, ODS, ODT, PDB, PRC, PDF, PPT, PPTX, RTF, TCR,
WPD, XLS, XLSX. Allows to alter a voice's parameters, including rate and pitch. the portable version available here.

>Optional secure download of installer at softonic



 however there are advantages to using dpscreenOCR as more suited to tables. **However note that dpscreenOCR reader missed out two words I would want** Bin and Doc. All will have understandable problems with un recognised symbols / lines
![image](https://user-images.githubusercontent.com/12754913/233661870-d82d9e14-a9c0-43e7-bd92-eca4692166bb.png)

For Read aloud I usually just use Windows
![image](https://user-images.githubusercontent.com/12754913/233667770-add3714b-0a44-46ed-9453-433150da8b97.png)
